packages/react-aria-components/stories/TagGroup.stories.tsx

```diff
@@ -52,3 +63,89 @@ TagGroupExample.argTypes = {
 function MyTag(props: TagProps) {
   return <Tag {...props} style={({isSelected}) => ({border: '1px solid gray', borderRadius: 4, padding: '0 4px', background: isSelected ? 'black' : '', color: isSelected ? 'white' : '', cursor: props.href ? 'pointer' : 'default'})} />;
 }
+
+
+export const TagGroupCollapsingExample = (props: TagGroupProps & {maxTags?: number}) => {
+  let {maxTags, ...otherProps} = props;
+  let [showAll, setShowAll] = useState(false);
+  let onAction = () => {
+    setShowAll(val => !val);
+  };
+  return (
+    <CollapsingCollection count={maxTags} showAll={showAll} onAction={onAction}>
+      <TagGroup {...otherProps}>
+        <Label>Categories</Label>
+        <TagList style={{display: 'flex', gap: 4}}>
+          <MyTag href="https://nytimes.com">News</MyTag>
+          <MyTag>Travel</MyTag>
+          <MyTag>Gaming</MyTag>
+          <MyTag>Shopping</MyTag>
+        </TagList>
+      </TagGroup>
+    </CollapsingCollection>
+  );
+};
+
+TagGroupCollapsingExample.args = {
+  selectionMode: 'none',
+  selectionBehavior: 'toggle'
+};
+
+TagGroupCollapsingExample.argTypes = {
+  selectionMode: {
+    control: {
+      type: 'inline-radio',
+      options: ['none', 'single', 'multiple']
+    }
+  },
+  selectionBehavior: {
+    control: {
+      type: 'inline-radio',
+      options: ['toggle', 'replace']
+    }
+  },
+  maxTags: {
+    control: {
+      type: 'number'
+    }
+  }
+};
+
+// Context for passing the count for the custom renderer
+interface ICollapseContext {
+  count?: number, onAction: () => void, showAll: boolean
+}
+let CollapseContext = React.createContext<ICollapseContext>({onAction: () => {}, showAll: false});
+
+function CollapsingCollection({children, count, onAction, showAll}) {
+  return (
+    <CollapseContext.Provider value={{count, onAction, showAll}}>
+      <CollectionRendererContext.Provider value={CollapsingCollectionRenderer}>
```

Comment:
Instead of a context you could potentially use function binding here, e.g. `value={collection => CollapsingCollectionRenderer(collection, props)`